app/orm-adapters/node-orm2.js

```diff
@@ -39,66 +39,68 @@ export default class NodeORM2Adapter extends ORMAdapter {
     return new AdapterModel(data);
   }
 
-  static idFor(record) {
-    return record.id;
+  static idFor(model) {
+    return model.record.id;
   }
 
-  static getAttribute(record, property) {
-    return record[property];
+  static getAttribute(model, property) {
+    return model.record[property];
   }
 
-  static setAttribute(record, property, value) {
-    return record[property] = value;
+  static setAttribute(model, property, value) {
+    return model.record[property] = value;
   }
 
-  static deleteAttribute(record, property) {
-    return record[property] = null;
+  static deleteAttribute(model, property) {
+    return model.record[property] = null;
   }
 
-  static getRelated(record, relationship) {
+  static getRelated(model, relationship) {
     return fromNode((cb) => {
-      record[`get${ upperFirst(relationship) }`](cb);
+      model.record[`get${ upperFirst(relationship) }`](cb);
     });
   }
 
-  static setRelated(record, relationship, relatedRecords) {
+  static setRelated(model, relationship, relatedRecords) {
     return fromNode((cb) => {
-      record[`set${ upperFirst(relationship) }`](relatedRecords, cb);
+      model.record[`set${ upperFirst(relationship) }`](relatedRecords, cb);
     });
   }
 
-  static addRelated(record, relationship, relatedRecord) {
+  static addRelated(model, relationship, relatedRecord) {
     return fromNode((cb) => {
-      record[`add${ upperFirst(relationship) }`](relatedRecord, cb);
+      model.record[`add${ upperFirst(relationship) }`](relatedRecord, cb);
     });
   }
 
-  static removeRelated(record, relationship, relatedRecord) {
+  static removeRelated(model, relationship, relatedRecord) {
     return fromNode((cb) => {
       let args = [ cb ];
       if (relatedRecord) {
         args.unshift([ relatedRecord ]);
       }
-      record[`remove${ upperFirst(relationship) }`](...args);
+      model.record[`remove${ upperFirst(relationship) }`](...args);
     });
   }
 
-  static saveRecord(record) {
-    return fromNode(record.save.bind(record));
+  static saveRecord(model) {
+    return fromNode(model.record.save.bind(model.record));
   }
 
-  static deleteRecord(record) {
-    return fromNode(record.remove.bind(record));
+  static deleteRecord(model) {
+    return fromNode(model.record.remove.bind(model.record));
   }
 
   static define(Model) {
     let attributes = {};
-    forIn(Model, (value, key) => {
-      if (value.isAttribute) {
-        attributes[this.serializeKey(key)] = this.typeForAttribute(value);
+    Model.eachAttribute((key, attribute) => {
+      if (attribute.isAttribute) {
+        attributes[key] = {
+          mapsTo: this.serializeKey(key),
+          type: this.typeForAttribute(attribute)
+        }
```

Comment:
Hm, looking at this again now, I think I was actually mistaken in my approach. If node-orm2 types are exposed through to the Denali models, theres no reason to map them to different values. You should just supply whatever type node-orm2 expects on your Denali model, and this adapter can just pass through. Although perhaps I'm missing something - it's late :wink:.

👍  on the `mapsTo` part.